### ISSUE TITLE
Não apareceram os campos CPF nem CNPJ

### ISSUE THREAD

**Issue by rafamuz** [claudiosanches/woocommerce-extra-checkout-fields-for-brazil#254]
Olá. Versão da WOOCOMMERCE: Versão 9.3.2. Instalei o Brazilian Market mas nao apareceram os campos CPF nem CNPJ.

**Comment by lucianoo6** [comment]
Estou enfrentando o mesmo problema é como o plug-in não estivesse ativado... de acordo com uma pesquisa breve o problema é com Wordpress 6.6.2

**Comment by rafamuz** [comment]
Sim, parece que abandonaram o projeto. pior que tentei encontrar o DEV para verificar umas duvidas e sugerir melhorias. procurei até nas redes sociais dele mandei mensagem e nunca respondeu, será que ele foi dessa para melhor? nao se sabe ao certo. sumiram! Já saíram várias versões e atualizações do W.P. que este plug-in já não contempla mais. o ideal também era que a comunidade fizesse o clone ou fork (nunca sei a diferença aqui) do projeto e fossem atualizando. não precisa necessariamente aguardarmos a boa vontade ou disponibilidade do dev principal, por se tratar de open-souce, quem puder colaborar seria uma boa para nós, pena eu não manjar NADA de PHP, senão eu mesmo dava uma mexida no código...

**Comment by lucianoo6** [comment]
Então depois de varias tentativas frustrada descobre que ele ficou incompatível com checkout moderno, funcionando apenas no checkout clássico. o checkout com WPBakery Page Builder também não funcionam. apenas adicionando o shortcode do woo pro checkout [woocommerce_checkout] em uma caixa de texto resolve porem volta ao horrível checkcout clássico.


![Captuuyurar](https://github.com/user-attachments/assets/f9232114-597e-4ca8-89ec-42bace0b1782)

**Comment by rhamses** [comment]
Passei por isso esse fim de semana também. Fui montando as pistas e este post ajudou bastante. Não sei exatamente quando isso aconteceu, mas a filosofia do checkout de bloco do WooCommerce agora é alinhada com o desenvolvimento dos blocos Gutenberg. 

Porém, existe maneiras de editar os campos deste novo checkout. De uma maneira mais "bruta" você pode seguir esse workshop que o time da [Woo fez ano passado explicando passo a passo e ainda dando um exemplo no github](https://developer.woocommerce.com/2023/08/07/extending-the-woocommerce-checkout-block-to-add-custom-shipping-options/?utm_source=pocket_shared). 

Em agosto desse ano eles lançaram um artigo sobre o [**Checkout API**](https://developer.woocommerce.com/roadmap/custom-checkout-fields/?utm_source=pocket_shared) que visa fazer essa interação mais próxima dos hooks que tinhamos no checkout antigo. 

E por fim, casando com este post de agosto, [achei essa documentação](https://github.com/woocommerce/woocommerce/blob/c07b7b8e9f638660aceed1fa1c6b76f29edc3fb3/docs/cart-and-checkout-blocks/additional-checkout-fields.md) no repositório deles explicando como adicionar campos direto no PHP, sem precisar criar bloco nenhum.

Este plugin aqui poderia ter. um PR fazendo essas atualizações porque mesmo depois que eu consegui inserir os campos que eu queria no formulário novo, ainda não consegui fazer o plugin do Melhor Envios reconhecer os campos e não me resolveu a questão das etiquetas. Mas só de ter a informação já é uma vitória. 

O código pra adicionar um campo no checkout novo via PHP segue dessa maneira:
```
woocommerce_register_additional_checkout_field(
			array(
				'id'            => 'namespace/nome-do-campo',
				'label'         => 'Nome do Campo',
				'optionalLabel' => 'Nome do Campo (opcional)',
				'location'      => 'address', // acredito que pode ser billing aqui tbm
				'required'      => true, // ou false
				'attributes'    => array(
					'autocomplete'     => '',
					'maxLength' 			 => '',
					'aria-describedby' => '',
					'aria-label'       => '',
					'pattern'          => '[0-9]{2}/[0-9]{2}/[0-9]{4}', 
					'title'            => '',
					'data-custom'      => ''
				),
				'validate_callback' => function ($field_value) {
					if (! is_email($field_value)) {
						return new WP_Error('error', 'Please ensure your alternative email matches the correct format.');
					}
				},
			),
		);
```

**Comment by webdados** [comment]
Pode usar isto https://github.com/Naked-Cat-Plugins/Simple-Checkout-Fields-Manager-for-WooCommerce---Validate-CPF-CNPJ

**Comment by tiago-sartor** [comment]
@rhamses O plugin tem um branch específico para desenvolvimento do suporte ao checkout de blocos ([veja aqui](https://github.com/claudiosanches/woocommerce-extra-checkout-fields-for-brazil/tree/blocks-support)). Se tiver interesse em contribuir, você pode abrir um Pull Request diretamente neste branch (ao fazer o fork você precisa **DESMARCAR** a caixa que diz "Copy the trunk branch only".
